### PR TITLE
dashboard-message-bus:add analytics message type

### PR DESCRIPTION
**Pull request by lsqproduction**
address #4842

### CODE REVIEW

**Review by fainashalts — APPROVED** [review]
Looks good to me! Nice use of type predicate!

File: packages/dashboard-message-bus/lib/message/types.ts [hunk 1]
```diff
@@ -41,6 +41,16 @@ export interface InvalidateMessage extends Message {
   payload: number;
 }
 
+/**
+ * Message intended to set analytic collection preference for user.
+ * The payload indicate whether or not the user opts in to share anonymous usage data with Truffle.
+ * This message payload is a truffle command request to update user config that should be forwarded.
+ */
+export interface SetAnalytics extends Message {
```
Comment (gnidan) [inline packages/dashboard-message-bus/lib/message/types.ts:49]: Looks like the other message types have `Message` in the name:

```suggestion
export interface SetAnalyticsMessage extends Message {
```

**Review by gnidan — CHANGES_REQUESTED** [review]
Looks like we should keep the name style consistent, at least in this PR. Otherwise, looks great!

File: packages/dashboard-message-bus/lib/message/types.ts (hunk 1 above)
Comment (gnidan) [reply packages/dashboard-message-bus/lib/message/types.ts:49]: (You'll have to update the other part of the code, also)

Comment (lsqproduction) [reply packages/dashboard-message-bus/lib/message/types.ts:49]: thank you @gnidan 
I will update it.

**Review by gnidan — APPROVED** [review]
Awesome, thanks!

**Pull request merged**